libs/expo/betterangels/src/lib/screens/AddEditClient/ContactInfo/MailingAddress.tsx

```diff
@@ -0,0 +1,126 @@
+import { Colors, Spacings } from '@monorepo/expo/shared/static';
+import {
+  BasicInput,
+  CardWrapper,
+  TextRegular,
+} from '@monorepo/expo/shared/ui-components';
+import { debounce } from '@monorepo/expo/shared/utils';
+import axios from 'axios';
+import { useCallback, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { TouchableOpacity, View } from 'react-native';
+import {
+  CreateClientProfileInput,
+  UpdateClientProfileInput,
+} from '../../../apollo';
+
+const apiUrl = process.env['EXPO_PUBLIC_API_URL'];
+
+export default function MailingAddress() {
+  const [suggestions, setSuggestions] = useState<any>([]);
+
+  const { setValue, watch } = useFormContext<
+    UpdateClientProfileInput | CreateClientProfileInput
+  >();
+
+  const mailingAddress = watch('mailingAddress') || '';
+
+  const onReset = () => {
+    setValue('mailingAddress', '');
+  };
+
+  const onSuggestionsSelect = (place: any) => {
+    setValue('mailingAddress', place.description.split(', ')[0]);
+    setSuggestions([]);
+  };
+
+  const searchPlacesInCalifornia = async (query: string) => {
```

Comment:
since we're using this in three places, can we move it to a shared util func and import?

Reply:
2 are identical. for interaction includes other logic. as it activates a modal.

done